lldb/source/Plugins/SymbolFile/DWARF/DWARFASTParserClang.cpp

```diff
@@ -1798,6 +1805,13 @@ DWARFASTParserClang::ParseStructureLikeDIE(const SymbolContext &sc,
   }
 
   if (def_die) {
+    if (auto [it, inserted] = dwarf->GetDIEToType().try_emplace(
```

Comment:
any chance of reducing/avoiding this duplication? Looks like the same code here as above? (refactor into a common function?)

Reply:
This part seems unnecessary as `UniqueDWARFASTType` map already handles it, right?

Reply:
For uniqueness, yes the unique map should already catch that (though, like we've seen, it can easily have bugs). However, it's still useful for preventing (infinite) recursion.